includes/admin/settings/class-wcmp-shipping-methods.php

```diff
@@ -86,20 +95,25 @@ private function addShippingMethod(string $key, string $value): void
         $this->shippingMethods[$key] = $value;
     }
 
+    /**
+     * @throws \Exception
+     */
     private function addShippingMethodsFromShippingZones(): void
     {
-        $shippingZones = WC_Shipping_Zones::get_zones();
-
-        foreach ($shippingZones as $shippingZone) {
+        foreach ($this->shippingZones as $shippingZone) {
             $zoneId = $shippingZone['id'] ?? $shippingZone['zone_id'];
 
             if (! $zoneId) {
                 continue;
             }
 
-            $zone = WC_Shipping_Zones::get_zone($zoneId);
+            $zone = new ShippingZone($shippingZone);
             /* @var WC_Shipping_Method[] $zoneShippingMethods */
-            $zoneShippingMethods = $zone->get_shipping_methods();
+            $zoneShippingMethods = $this->shippingZones[$zone->get_id()]['shipping_methods'] ?? null;
```

Comment:
je kunt ook die `$zone->get_shipping_methods()` hier op het eind schrijven in plaats van die `?? null`. Misschien is het nog beter om dit stukje in een losse functie te trekken.